lib/crypto/crypto_file.c

```diff
@@ -43,6 +45,14 @@ crypto_file_init_keys(void)
 	    256)) == NULL)
 		goto err0;
 
+	if (atexit(crypto_file_atexit)) {
+		warnp("Could not initialize atexit");
+
+		/* Call manually because we've already allocated decr_aes_cache. */
```

Comment:
Just make this a "goto err1" and add "err1: rwhashtab_free(decr_aes_cache);".  No point diverging from normal style, and it's more obviously correct if a failed _init is cleaning up precisely what it allocated up to when it failed.